app.js

```diff
@@ -1,59 +1,135 @@
+        td3.innerHTML = '₱ ' + clients[row].balance;
+        let a = document.createElement("a");
+        td4.append(a);
+        a.innerHTML = "Transaction";
+        a.href = "#";
 
-function clearInputFields() {
-    document.getElementById('accntName').value = "";
-    document.getElementById('accntNo').value = "";
-    document.getElementById('initDep').value = "";
+    })
 }
 
 function addClient(){   
-
+    // console.log("Hello World!")
     let newClient = {};
+    
+    try {
+        if(accntName.value == "" ) throw "Account Name is requied!";
+        if(accntNo.value == "") throw "Account Number is required";
+        if(initDep.value == "") throw "Initial Balance is required";        
+    } catch (err) {
+        alert(err);
+        return;
+    }
+
+    let x = Object.keys(clients);
+    for(let i = 0; i < x.length; i++){
+        let accName = clients[i].accountName;
+        let accNo = clients[i].accountNumber;
+        let strAccName = accName.toUpperCase();
+        let strAccntName = accntName.value.toUpperCase();
+        if(strAccName === strAccntName){
+            alert("Client Already Exist.");
+            return;
+        }
+        if(accNo === accntNo.value){
+            alert("Account Number not available.");
+            return;
+        }
+    }
+    newClient.accountName = accntName.value;
+    newClient.accountNumber = accntNo.value;
+    // accntType = typeOfAccnt.value;
+    newClient.typeOfAccount = typeOfAccnt.value;
+
     let bal = initDep.value;
-    let accType = radios.value;
     let formattedBal = bal.replace(/,/gi, "");
     let newBal = formattedBal.replace(/\d(?=(?:\d{3})+$)/g, '$&,');
-
+    newClient.balance = newBal;
+    let today = new Date;
+    let dd = String(today.getDate()).padStart(2, '0');
+    let mm = String(today.getMonth() + 1).padStart(2, '0')
+    let yyyy = today.getFullYear();
+    today = dd + '/' + mm + '/' + yyyy;
+    newClient.transactionLog = [{
+        accountNumber:`${newClient.accountNumber}`,
+        transaction:"Created " + `${newClient.accountName}` + " with initial deposit of " + `${newClient.balance}`,
+        date: today
+    }]
+    clients.push(newClient);
+    let client = JSON.stringify(newClient);
+    localStorage.setItem(`${id}`, client);
+    id++;
+    
+    
     let table = document.getElementById("clientList");
-    let row = table.insertRow();
+    let tbody = document.createElement("tbody");
+    table.appendChild(tbody);
+    let row = tbody.insertRow();
    
     let cName = row.insertCell(0);
     let cNo = row.insertCell(1);
     let cType = row.insertCell(2);
     let cBal = row.insertCell(3);
     let tLog = row.insertCell(4);
 
-    newClient.accountName = accntName.value;
-    newClient.accountNumber = accntNo.value;
-    newClient.balance = newBal; 
-
     cName.innerHTML = newClient.accountName;
     cNo.innerHTML = newClient.accountNumber;
-    cType.innerHTML = accType;
+    cType.innerHTML = newClient.typeOfAccount;
```

Comment:
` let accType = radios.value;`
` cType.innerHTML = accType;`

_change line 122 to these codes_